theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,118 @@
+# Custom Server Side Ad Insertion
+
+THEOplayer provides a way to integrate with third-party advertisement providers,
+and have them report their ads and ad-related events through the THEOplayer APIs.
+
+## Android SDK
+
+### Prerequisites
+
+Your THEOplayer SDK needs to be on version 7.6.0 or higher.
+
+### Integration ID
+
+To make use of the custom server side ad insertion, first you need an integration ID which will identify the provided sources and ads to the specific ad provider.
+
+### CustomSsaiDescriptionSerializer
+
+Register a CustomSsaiDescriptionSerializer for the specific integration to be able to serialize/deserialize the
+CustomSsaiDescription that will be provided in the SourceDescription.
+
+```kotlin
+CustomSsaiDescriptionRegistry.register("integrationID", object : CustomSsaiDescriptionSerializer {
+
+    /**
+     * Reads a CustomSsaiDescription from the given JSON string.
+     *
+     * The customIntegration property in CustomSsaiDescription of the returned object
+     * **must** match the integration with which this serializer was registered.
+     */
+    override fun fromJson(json: String): CustomSsaiDescription {
+
+    }
+
+    /**
+     * Writes the given CustomSsaiDescription as a JSON string.
+     *
+     * The returned JSON string **must** have a string property named `integration`
+     * that contains the value of customIntegration property in CustomSsaiDescription.
+     */
+    override fun toJson(value: CustomSsaiDescription): String {
+
+    }
+})
+```
+
+### ServerSideAdIntegrationFactory
+
+You would also need to register a ServerSideAdIntegrationFactory that will receive a ServerSideAdIntegrationController.
+The controller can be used to create the THEOplayer AdBreak/Ad objects using the AdInit data.
+Additionally, it can be used to keep the player's state up-to-date such as when an Ad begins or ends.
+
+The ServerSideAdIntegrationFactory requires an implementation of the ServerSideAdIntegrationHandler interface.
+The handler will provide few callbacks that can be overridden which allows you to setup/clear the integration as needed.
+
+```kotlin
+theoplayerView?.player?.ads?.registerServerSideIntegration("integrationID") { controller ->
+    object : ServerSideAdIntegrationHandler {
+        /**
+         * Handler which will be called when a new source is loaded into the player.
```

Comment:
This is just repeating [our API documentation](https://www.theoplayer.com/docs/theoplayer/v8/api-reference/android/com/theoplayer/android/api/ads/ServerSideAdIntegrationHandler.html). That's not very useful.

For a how-to guide, we should try to show some actual example code, e.g. create an ad with `controller.createAd()`, start it with `beginAd()`, etc. We could also link to some real-world usages, such as the [Yospace connector](https://github.com/THEOplayer/android-connector/tree/master/connectors/yospace) and [Uplynk connector](https://github.com/THEOplayer/android-connector/tree/master/connectors/uplynk).